src/lib/components/ui/TextField/TextField.jsx

```diff
@@ -6,7 +6,31 @@ import styles from './TextField.scss';
 
 export const TextField = (props) => {
   const SMALL_INPUT_SIZE = 10;
-  const HAS_SMALL_INPUT = props.inputSize && props.inputSize <= SMALL_INPUT_SIZE;
+
+  const setCustomInputSizeByType = (inputType, inputSize, max) => {
```

Comment:
I do no personally prefer to have this kind of functions if they are used only once. In this case, you can set value of `customInputeSize` directly without function. If you still want to do that (not only here, but globally), I would prefer to move it out of component function (e.g. before function or into helper file). Advantage of this is that you can use [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo) to not to calculate those values on every time component rerenders

Reply:
I favour this kind of functions if they improve readability of the code. Without this function, the constant `customInputSize` would look like this:

```js
const customInputSize = (inputType === 'number' && max)
  ? max.toString().length
  : (inputSize || null);
```

(Not so bad, surprisingly…!)

Or with `let` which I generally try to avoid:

```js
let customInputSize = null;

if (inputType === 'number' && max) {
  customInputSize = max.toString().length;
} else if (inputSize) {
  customInputSize = inputSize;
}
```

Or did you mean just to make the function anonymous?

Reply:
You can leave those functions as they now, just move them out of `export const TextField = (props) => {}` function. I think that we could/should to this everywhere. Somewhere I read that it is performing better when those helpers are not part of component function (e.g.  ` const TextField = (props) => {}`) because they are created everytime `TextField(props)` is called. Plus you can use `useMemo` (as mentioned above) not to call those helpers if props does not change.

Reply:
For the record: on Slack we decided to leave it as it is. We'll get back to the topic of helper functions at the next meeting.